reana_db/models.py

```diff
@@ -286,6 +286,58 @@ class JobStatus(enum.Enum):
     queued = 5
 
 
+class UserWorkflowSession(Base, Timestamp):
+    """User Workflow Session table."""
+
+    __tablename__ = "user_workflow_session"
+    __table_args__ = {"schema": "__reana"}
+
+    user_id = Column(UUIDType, ForeignKey("__reana.user_.id_"), nullable=False)
+    workflow_id = Column(UUIDType, ForeignKey("__reana.workflow.id_"), nullable=True)
+    session_id = Column(
+        UUIDType, ForeignKey("__reana.interactive_session.id_"), primary_key=True
+    )
+
+    def __repr__(self):
+        """User Workflow Session string representation."""
+        return "<UserWorkflowSession {} {} {}>".format(
+            self.user_id, self.session_id, self.workflow_id
+        )
+
+
+class InteractiveSessionType(enum.Enum):
```

Comment:
I now realise we have a little problem with this enum.

We already have a [`INTERACTIVE_SESSION_TYPES`](https://github.com/reanahub/reana-commons/blob/e6cdbedb62b21cc20407512e7d0a09eeb28f9481/reana_commons/config.py#L261:L261) config variable in REANA-Commons. We could completely retire it an replace it with this new enum if it were not because of [REANA-Client depending on it](https://github.com/reanahub/reana-client/blob/1e69bc3a8e48769b75b6f8ff267c7974050411ac/reana_client/cli/workflow.py#L1236).

We have two options:
1. Keep the enum, remove `INTERACTIVE_SESSION_TYPES` and therefore "disregard" the [client-side validation in REANA-Client](https://github.com/reanahub/reana-client/blob/1e69bc3a8e48769b75b6f8ff267c7974050411ac/reana_client/cli/workflow.py#L1236) and deal with it in either of the following ways
     1. Letting errors be triggered and rely on the server-side message to tell the user what are the possible values for the interactive sessions
     2. Offer the `InteractiveSessionType` through REST API so we can achieve the client-side validation after a first request retrieving config
2. Keep the `INTERACTIVE_SESSION_TYPES` and all kind of `_TYPES` variables in REANA-Commons because we value the CLI client-side validation in the short term


I think that `1. ii.` is the best option, as it could lead to a future endpoint that can return all of these kind of configs aggregated to both R-Client and R-UI can benefit from it. If we decide on it, for now in R-Client we can live without client-side validation and create and issue to implement the previously mentioned mechanism.

Reply:
I've created an [issue](https://github.com/reanahub/reana-client/issues/430) to address the refactoring in REANA-Client and REANA-Commons, for the rest services refactored the code to use the new enum